### PR TITLE
Roleplay essentials: flavor text & records

### PR DESCRIPTION
**Pull request by Ephemeralis**
## About The Pull Request

This gets probably the last major feature before version 0 can go live into place: *flavour preferences*.

It is a mixture of stuff ported from NovaSector and new code.

Notable additions/differences:
- **Flavour text and species information is output in chat versus a new TGUI window.** This started off as a concession towards the effort involved in making an entirely new TGUI window design, but then started to gain some aesthetic merit in its own right.
- **Relevant character preferences have been split out into their own Lore page in the character creator.** Most of us are used to Nova's way of doing things (aka declaring every new preference as "non-contextual" and shoving it in the same category.
- **Silicon-specific preferences.** Cyborgs have their own description **and** species/model fields, so you can now feasibly play a cyborg character who swaps out into an IPC/synthetic shell mid-round with an entirely unique description that you've written for them. This also includes headshot images.
- **Basic player-written record support.** Kind of shitty at the moment but they appear in both security and medical consoles, PDA apps forthcoming. Exploitables are renamed to 'classified records' and guidance is given in the character creation screen about writing interesting ones that aren't just some variant of 'my character has a stash of hardcore pornography involving a particular species of desk fern that they're really bashful about'.

https://github.com/user-attachments/assets/0ef7022b-be53-4155-a209-90bad0af645a

https://github.com/user-attachments/assets/6f89a15e-5214-4bb5-ae47-481fe3709af5

In preparation for further customization options linked to preferences later down the line, I've also cut out a `PreferenceSingle` react element that references a single preference by its chosen savefile key, allowing people who aren't very acquainted with react/typescript (me and the rest of the fucking planet) to create new pages for preferences that aren't wrapped three layers deep in autogenerated wizardry. Mothblocks did an amazing job on the auto-preference stuff, and this gets us the best of both worlds.

## Changelog

:cl: yooriss
add: Bespoke in-chat flavour text via examines. Also includes custom species information in a new, separate dialogue!
add: Flavor text, records and other important roleplay-centric character information can now be set via the Lore panel in the character creator.
add: Player-written records are viewable in security and medical records consoles.
code: New PreferenceSingle react element to make creating TGUI interfaces for new preferences significantly easier (where specific sorting/ordering is needed)
/:cl:

### CODE REVIEW

File: code/datums/records/manifest.dm [hunk 1]
```diff
@@ -15,7 +15,7 @@ GLOBAL_DATUM_INIT(manifest, /datum/manifest, new)
 		if(readied_player.new_character)
 			log_manifest(readied_player.ckey, readied_player.new_character.mind, readied_player.new_character)
 		if(ishuman(readied_player.new_character))
-			inject(readied_player.new_character)
+			inject(readied_player.new_character, readied_player.client)
```
Comment (CliffracerX) [inline code/datums/records/manifest.dm:18]: Doppler edit marker needed here

File: code/datums/records/manifest.dm [hunk 2]
```diff
@@ -131,6 +131,7 @@ GLOBAL_DATUM_INIT(manifest, /datum/manifest, new)
 		// Locked specifics
 		locked_dna = record_dna,
 		mind_ref = person.mind,
+		age_chronological = person_client?.prefs.read_preference(/datum/preference/numeric/chronological_age),
```
Comment (CliffracerX) [inline code/datums/records/manifest.dm:134]: Ditto

File: code/datums/records/record.dm [hunk 3]
```diff
@@ -152,10 +165,12 @@
 	/// Locked specific
 	datum/dna/locked_dna,
 	datum/mind/mind_ref,
+	age_chronological = 18,
```
Comment (CliffracerX) [inline code/datums/records/record.dm:168]: Ditto

File: code/datums/records/record.dm [hunk 4]
```diff
@@ -152,10 +165,12 @@
...
 )
 	. = ..()
 	src.locked_dna = locked_dna
 	src.mind_ref = WEAKREF(mind_ref)
+	src.age_chronological = age_chronological
```
Comment (CliffracerX) [inline code/datums/records/record.dm:173]: Ditto

File: code/modules/admin/admin_verbs.dm [hunk 5]
```diff
@@ -553,7 +553,7 @@ ADMIN_VERB(spawn_debug_full_crew, R_DEBUG, "Spawn Debug Full Crew", "Creates a f
 		// Finally, ensure the minds are tracked and in the manifest.
 		SSticker.minds += character.mind
 		if(ishuman(character))
-			GLOB.manifest.inject(character)
+			GLOB.manifest.inject(character, character.client)
```
Comment (CliffracerX) [inline code/modules/admin/admin_verbs.dm:556]: Ditto

File: code/modules/admin/verbs/admingame.dm [hunk 6]
```diff
@@ -277,7 +277,7 @@ ADMIN_VERB(respawn_character, R_ADMIN, "Respawn Character", "Respawn a player th
 	if(!record_found && (new_character.mind.assigned_role.job_flags & JOB_CREW_MEMBER))
 		//Power to the user!
 		if(tgui_alert(new_character,"Warning: No data core entry detected. Would you like to announce the arrival of this character by adding them to various databases, such as medical records?",,list("No","Yes")) == "Yes")
-			GLOB.manifest.inject(new_character)
+			GLOB.manifest.inject(new_character, new_character.client)
```
Comment (CliffracerX) [inline code/modules/admin/verbs/admingame.dm:280]: Ditto

File: code/modules/mob/dead/new_player/new_player.dm [hunk 7]
```diff
@@ -243,7 +243,7 @@
 		SSquirks.AssignQuirks(humanc, humanc.client)
 
 	if(humanc) // Quirks may change manifest datapoints, so inject only after assigning quirks
-		GLOB.manifest.inject(humanc)
+		GLOB.manifest.inject(humanc, humanc.client)
```
Comment (CliffracerX) [inline code/modules/mob/dead/new_player/new_player.dm:246]: Ditto

File: code/modules/mob/living/silicon/robot/examine.dm [hunk 8]
```diff
@@ -1,8 +1,20 @@
+#define ADD_NEWLINE_IF_NECESSARY(list) if(length(list) > 0 && list[length(list)]) { list += "" }
```
Comment (CliffracerX) [inline code/modules/mob/living/silicon/robot/examine.dm:1]: Ditto *probably*?  Or migrate this to one of our defines files?

File: tgui/packages/tgui-panel/styles/tgchat/chat-dark.scss [hunk 9]
```diff
@@ -1018,6 +1018,25 @@ em {
   flex-grow: 1;
 }
 
+// Larger images by text, used by flavor-text examines. No flex shenanigans.
```
Comment (CliffracerX) [inline tgui/packages/tgui-panel/styles/tgchat/chat-dark.scss:1021]: Should demarkate this with a note about it being a doppler edit as well

File: tgui/packages/tgui-panel/styles/tgchat/chat-light.scss [hunk 10]
```diff
@@ -1036,6 +1036,25 @@ h2.alert {
   flex-grow: 1;
 }
 
+// Larger images by text, used by flavor-text examines. No flex shenanigans.
```
Comment (CliffracerX) [inline tgui/packages/tgui-panel/styles/tgchat/chat-light.scss:1039]: Ditto

File: tgui/packages/tgui/interfaces/MedicalRecords/RecordView.tsx [hunk 11]
```diff
@@ -87,7 +92,7 @@ export const MedicalRecordView = (props) => {
             <LabeledList.Item label="Job">
               <EditableText field="job" target_ref={crew_ref} text={rank} />
             </LabeledList.Item>
-            <LabeledList.Item label="Age">
+            <LabeledList.Item label="Physical Age">
```
Comment (CliffracerX) [inline tgui/packages/tgui/interfaces/MedicalRecords/RecordView.tsx:95]: Ditto

File: tgui/packages/tgui/interfaces/PreferencesMenu/CharacterPreferenceWindow.tsx [hunk 12]
```diff
@@ -8,6 +8,7 @@ import { AntagsPage } from './AntagsPage';
 import { PreferencesMenuData } from './data';
 import { JobsPage } from './JobsPage';
 import { LoadoutPage } from './loadout/index';
+import { LorePage } from './LorePage';
```
Comment (CliffracerX) [inline tgui/packages/tgui/interfaces/PreferencesMenu/CharacterPreferenceWindow.tsx:11]: Ditto

File: tgui/packages/tgui/interfaces/PreferencesMenu/CharacterPreferenceWindow.tsx [hunk 13]
```diff
@@ -20,6 +21,7 @@ enum Page {
   Species,
   Quirks,
   Loadout,
+  Lore,
```
Comment (CliffracerX) [inline tgui/packages/tgui/interfaces/PreferencesMenu/CharacterPreferenceWindow.tsx:24]: Ditto

File: tgui/packages/tgui/interfaces/PreferencesMenu/CharacterPreferenceWindow.tsx [hunk 14]
```diff
@@ -82,6 +84,10 @@ export const CharacterPreferenceWindow = (props) => {
       pageContents = <LoadoutPage />;
       break;
 
+    case Page.Lore:
```
Comment (CliffracerX) [inline tgui/packages/tgui/interfaces/PreferencesMenu/CharacterPreferenceWindow.tsx:87]: Ditto

File: tgui/packages/tgui/interfaces/PreferencesMenu/data.ts [hunk 15]
```diff
@@ -138,6 +138,7 @@ export type PreferencesMenuData = {
       random_body: RandomSetting;
       [otherKey: string]: unknown;
     };
+    doppler_lore: Record<string, unknown>;
```
Comment (CliffracerX) [inline tgui/packages/tgui/interfaces/PreferencesMenu/data.ts:141]: Ditto

File: tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/base.tsx [hunk 16]
```diff
@@ -17,6 +17,7 @@ import {
   NumberInput,
   Slider,
   Stack,
+  TextArea,
```
Comment (CliffracerX) [inline tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/base.tsx:20]: Ditto, but comments might not work on this so if you can't no big deal

File: tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/age.tsx [hunk 17]
```diff
@@ -2,5 +2,7 @@ import { Feature, FeatureNumberInput } from '../base';
 
 export const age: Feature<number> = {
   name: 'Age',
+  description:
```
Comment (CliffracerX) [inline tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/character_preferences/age.tsx:5]: Ditto, though see above; might not be able to comment in the technical list here

File: tgui/packages/tgui/interfaces/SecurityRecords/RecordView.tsx [hunk 18]
```diff
@@ -146,7 +151,7 @@ const RecordInfo = (props) => {
             <LabeledList.Item label="Job">
               <EditableText field="rank" target_ref={crew_ref} text={rank} />
             </LabeledList.Item>
-            <LabeledList.Item label="Age">
+            <LabeledList.Item label="Physical Age">
```
Comment (CliffracerX) [inline tgui/packages/tgui/interfaces/SecurityRecords/RecordView.tsx:154]: Ditto

**Review by CliffracerX — CHANGES_REQUESTED** [review]
This is just about ready, just a few things need properly demarkated with doppler edit indicators so it's easier to spot our stuff in future merges.

**Review by honkpocket — APPROVED** [review]

**Review by CliffracerX — APPROVED** [review]
Looks ready to go now!

**Pull request merged**